### PR TITLE
Feature/search book

### PR DESCRIPTION
**Pull request by meerimb**
added the function of searching and sorting books

### CODE REVIEW

File: src/main/java/kg/peaksoft/ebookm1/exceptions/ErrorDTO.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+@Getter
+@Setter
+public class ErrorDTO {
+
+    private Date timestamp;
+    private String message;
+    private String details;
+
+    public ErrorDTO(String message,
```
Comment (nurlan312) [inline src/main/java/kg/peaksoft/ebookm1/exceptions/ErrorDTO.java:16]: Deleted constructor

File: src/main/java/kg/peaksoft/ebookm1/exceptions/ErrorDTO.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package kg.peaksoft.ebookm1.exceptions;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Date;
+
+@Getter
+@Setter
```
Comment (nurlan312) [inline src/main/java/kg/peaksoft/ebookm1/exceptions/ErrorDTO.java:9]: Add @RequiredArgsConstructor

File: src/main/java/kg/peaksoft/ebookm1/exceptions/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(ResourceNotFoundException.class)
+    public ResponseEntity<ErrorDTO> handleResourceNotFoundException(ResourceNotFoundException exception,
+                                                                    WebRequest request) {
+        ErrorDTO errorDTO = new ErrorDTO(exception.getMessage(),
```
Comment (nurlan312) [inline src/main/java/kg/peaksoft/ebookm1/exceptions/GlobalExceptionHandler.java:15]: Corrected to ErrorDTO errorDTO = new ErrorDTO(new Date(), exception.getMessage(),
                request.getDescription(false));

File: src/main/java/kg/peaksoft/ebookm1/exceptions/GlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                request.getDescription(false));
+        return new ResponseEntity<>(errorDTO, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(MyException.class)
+    public ResponseEntity<ErrorDTO> handleMeException(MyException exception, WebRequest request) {
+        ErrorDTO errorDTO = new ErrorDTO(exception.getMessage(),
```
Comment (nurlan312) [inline src/main/java/kg/peaksoft/ebookm1/exceptions/GlobalExceptionHandler.java:22]: The same

File: src/main/java/kg/peaksoft/ebookm1/exceptions/GlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                request.getDescription(false));
+        return new ResponseEntity<>(errorDTO, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<ErrorDTO> handleGlobalException(Exception exception, WebRequest request) {
+        ErrorDTO errorDTO = new ErrorDTO(exception.getMessage(),
```
Comment (nurlan312) [inline src/main/java/kg/peaksoft/ebookm1/exceptions/GlobalExceptionHandler.java:29]: The same

File: src/main/java/kg/peaksoft/ebookm1/exceptions/ErrorDTO.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Date;
+
+@Getter
+@Setter
+@AllArgsConstructor
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/ebookm1/exceptions/ErrorDTO.java:11]: Why @AllArgsConstructor?

**Review by ChyngyzIshen — COMMENTED** [review]
comment

**Pull request merged**